packages/webpack-plugin/src/plugin.ts

```diff
@@ -0,0 +1,463 @@
+                    assetsModules.set(module.resource, module);
+                }
+            }
+        });
+
+        /**
+         * @remove
+         * This part supports old loaders and should be removed
+         */
+        if (this.options.assetsMode === 'loader') {
+            compilation.hooks.optimizeDependencies.tap(StylableWebpackPlugin.name, () => {
+                for (const module of stylableModules) {
+                    const connections = moduleGraph.getOutgoingConnections(module);
+                    for (const connection of connections) {
+                        if (
+                            !isAssetModule(connection.module) &&
+                            isLoadedWithKnownAssetLoader(connection.module)
+                        ) {
+                            connection.setActive(false);
+                        }
+                    }
+                }
+            });
+        }
+
+        /**
+         *  After we have the initial chunks we can calculate the depth and usage of each stylesheet
+         */
+        compilation.hooks.afterChunks.tap({ name: StylableWebpackPlugin.name, stage: 0 }, () => {
+            for (const module of stylableModules) {
+                module.buildMeta.stylable.isUsed = findIfStylableModuleUsed(module, compilation);
+                module.buildMeta.stylable.depth = calcDepth(module, moduleGraph);
+            }
+        });
+
+        compilation.hooks.afterChunks.tap(StylableWebpackPlugin.name, () => {
+            const optimizer = this.stylable.optimizer!;
+            const optimizeOptions = this.options.optimize;
+            const sortedModules = getSortedModules(stylableModules);
+
+            const {
+                usageMapping,
+                namespaceMapping,
+                namespaceToFileMapping,
+            } = createOptimizationMapping(sortedModules, optimizer);
+
+            if (!this.options.unsafeMuteDiagnostics.DUPLICATE_MODULE_NAMESPACE) {
+                reportNamespaceCollision(namespaceToFileMapping, compilation.errors);
+            }
+
+            for (const module of sortedModules) {
+                const buildMeta = getStylableBuildMeta(module);
+                const { css, exports, globals } = buildMeta;
+
+                const ast = parse(css);
+
+                optimizer.optimizeAst(
+                    optimizeOptions,
+                    ast,
+                    usageMapping,
+                    this.stylable.delimiter,
+                    exports,
+                    globals
+                );
+
+                buildMeta.css = optimizeOptions.minify
+                    ? optimizer.minifyCSS(ast.toString())
+                    : ast.toString();
+
+                if (optimizeOptions.shortNamespaces) {
+                    buildMeta.namespace = namespaceMapping[buildMeta.namespace];
+                }
+            }
+        });
+    }
+    private chunksIntegration(
```

Comment:
rename method with something that says it's about injecting the css assets

Reply:
can you suggest a name. I like that separation of modules and chunks....